Mathlib/Algebra/SkewMonoidAlgebra/Basic.lean

```diff
@@ -423,8 +446,386 @@ theorem sum_congr {f : SkewMonoidAlgebra k G} {M : Type*} [AddCommMonoid M] {g
     (h : ∀ x ∈ f.support, g₁ x (f.coeff x) = g₂ x (f.coeff x)) :
     f.sum g₁ = f.sum g₂ := Finset.sum_congr rfl h
 
+theorem induction_on {p : SkewMonoidAlgebra k G → Prop} (f : SkewMonoidAlgebra k G)
```

Comment:
The names of the parameters `h0`, `hM`, `hadd` have to be provided in the syntax `induction … with`. Therefore, I suggest that you call them `zero`, `single`, `add`.